### DISCUSSION TITLE
Feature request:how to test for watch function in tsx file?

### DISCUSSION THREAD

**Discussion by 916871270** [vuejs/test-utils#1993] [General]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by 916871270** [comment]
setup(props) {
      const isShow = ref(false)
      watch(props, (newProps) => {
          isShow.value = newProps.show
      })
}

**Comment by cexbrayat** [comment]
Hi @916871270 

It's hard to help you without more context, but you can probably write something like this:

```ts
const wrapper = mount(MyComponent);
expect(wrapper.vm.isShow).toBe(false);
await wrapper.setProps({ show: true });
expect(wrapper.vm.isShow).toBe(true);
```

A better test would be to test what really changes in the template when `isShow` has a new value.
